### PR TITLE
[2819] Initialize Deck representation support

### LINKED ISSUE #2819: Contribute the first version of the Deck representation
* [x] **I have checked that this feature has not yet been suggested by someone else.**

### Expected behavior

have a first step containing:

* A Deck representation description
* Stereotypes to create the deck view example
* Make it possible to create a Deck Representation based on a deck description
* Render the Deck representation with the react-trello component.

...

### Cuttings backs

* There will be no editing in this first contribution.
* The toolbar will be present but not functional yet.
...

### Potential side effects

...

### Describe alternatives you've considered

...

### PR DESCRIPTION
**Pull request by florianbarbin**
* This commit allows to perform the minimal use case: 
    * Create a Deck view description
    * Initalize a deck view description from the provided stereotype 
    * Render a new Deck Representation based the deck description 
    * Display the deck representation in the front-end, based on the react-trello component.
* There is no editing from the front-end.
* The Deck representation is refreshed if the semantic model has changed.

Bug: https://github.com/eclipse-sirius/sirius-web/issues/2819

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [x] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [x] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [x] Have the relevant issues been added to the same project and milestone as the pull request?
- [x] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [x] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [x] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?

1. Create a new Blank Studio
2. Create new Deck Description from the provided stereotype
3. Import the given task model [Task Model.zip](https://github.com/eclipse-sirius/sirius-web/files/13674567/Task.Model.zip)
4. Create a new Deck Representation on the project.

_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/deck/backend/sirius-components-collaborative-deck/pom.xml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.1.2</version>
+		<relativePath /> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>org.eclipse.sirius</groupId>
+	<artifactId>sirius-components-collaborative-deck</artifactId>
+    <version>2023.12.1</version>
```
Comment (frouene) [inline packages/deck/backend/sirius-components-collaborative-deck/pom.xml]: Must be updated to last sirius version for now 2023.12.3

File: packages/deck/backend/sirius-components-deck-graphql/pom.xml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.1.2</version>
+		<relativePath /> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>org.eclipse.sirius</groupId>
+	<artifactId>sirius-components-deck-graphql</artifactId>
+    <version>2023.12.1</version>
```
Comment (frouene) [inline packages/deck/backend/sirius-components-deck-graphql/pom.xml]: The version need to be updated

File: packages/deck/backend/sirius-components-deck/pom.xml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.1.2</version>
+		<relativePath /> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>org.eclipse.sirius</groupId>
+	<artifactId>sirius-components-deck</artifactId>
+	<version>2023.12.1</version>
```
Comment (frouene) [inline packages/deck/backend/sirius-components-deck/pom.xml]: The version need to be updated

File: packages/deck/frontend/sirius-components-deck/package.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    "@material-ui/icons": "4.11.3",
+    "@xstate/react": "1.6.3",
+    "graphql": "16.8.0",
+    "react": "17.0.2",
+    "react-dom": "17.0.2",
+    "xstate": "4.32.1"
+  },
+  "devDependencies": {
+    "@apollo/client": "3.8.1",
+    "@eclipse-sirius/sirius-components-core": "*",
```
Comment (frouene) [inline packages/deck/frontend/sirius-components-deck/package.json:45]: doesn't this dependency need to be declared in peerDependencies?

File: packages/deck/frontend/sirius-components-deck/src/representation/DeckRepresentation.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  if (complete) {
+    content = (
+      <div className={classes.complete}>
+        <Typography variant="h5" align="center">
+          The Deck does not exist anymore
+        </Typography>
+      </div>
+    );
+  } else if (deck) {
+    const lanes = deck?.lanes ?? [];
```
Comment (frouene) [inline packages/deck/frontend/sirius-components-deck/src/representation/DeckRepresentation.tsx]: ? is useless, deck is always definied here

File: packages/sirius-web/backend/sirius-web-sample-application/src/main/java/org/eclipse/sirius/web/sample/task/configuration/view/ViewDeckDescriptionBuilder.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        DeckDescription deckDescription = this.createDeckDescription();
+        LaneDescription laneDescription = this.createLaneDescription();
+        CardDescription cardDescription = this.createCardDescription();
+        deckDescription.getLaneDescriptions().add(laneDescription);
+        laneDescription.getOwnedCardDescriptions().add(cardDescription);
+        view.getDescriptions().add(deckDescription);
+    }
+
+    private CardDescription createCardDescription() {
+        return this.deckBuilders.newCardDescription()//
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web-sample-application/src/main/java/org/eclipse/sirius/web/sample/task/configuration/view/ViewDeckDescriptionBuilder.java]: You should be able to keep line wrap without add //

**Review by frouene — CHANGES_REQUESTED** [review]

File: packages/deck/backend/sirius-components-collaborative-deck/pom.xml (hunk 1 above)
Comment (florianbarbin) [reply packages/deck/backend/sirius-components-collaborative-deck/pom.xml]: done

File: packages/deck/backend/sirius-components-deck-graphql/pom.xml (hunk 2 above)
Comment (florianbarbin) [reply packages/deck/backend/sirius-components-deck-graphql/pom.xml]: done

File: packages/deck/backend/sirius-components-deck/pom.xml (hunk 3 above)
Comment (florianbarbin) [reply packages/deck/backend/sirius-components-deck/pom.xml]: done

File: packages/deck/frontend/sirius-components-deck/package.json (hunk 4 above)
Comment (florianbarbin) [reply packages/deck/frontend/sirius-components-deck/package.json:45]: yes it does

File: packages/deck/frontend/sirius-components-deck/src/representation/DeckRepresentation.tsx (hunk 5 above)
Comment (florianbarbin) [reply packages/deck/frontend/sirius-components-deck/src/representation/DeckRepresentation.tsx]: yes, fixed.

File: packages/sirius-web/backend/sirius-web-sample-application/src/main/java/org/eclipse/sirius/web/sample/task/configuration/view/ViewDeckDescriptionBuilder.java (hunk 6 above)
Comment (florianbarbin) [reply packages/sirius-web/backend/sirius-web-sample-application/src/main/java/org/eclipse/sirius/web/sample/task/configuration/view/ViewDeckDescriptionBuilder.java]: done

**Review by frouene — APPROVED** [review]

**Comment by frouene** [comment]
Failed tests are not related to the PR

**Pull request merged**